src/net_processing.cpp

```diff
@@ -4423,6 +4431,23 @@ class CompareInvMempoolOrder
 };
 }
 
+bool PeerManagerImpl::SetupAddressRelay(CNode& node)
+{
+    // We don't participate in addr relay with outbound block-relay-only
+    // connections to prevent providing adversaries with the additional
+    // information of addr traffic to infer the link.
+    if (node.IsBlockOnlyConn()) return false;
+
+    PeerRef peer = GetPeerRef(node.GetId());
```

Comment:
no particular reason _not_ to, I'm wondering what the difference is... is this suggestion slightly less overhead because `GetPeerRef` would construct an additional `shared_ptr`, which would then have to update the internal state used to coordinate. VS if we just pass in a reference (I'm thinking `Peer&`?), there's less bookkeeping since the lifetime of the reference would just be managed in the normal program execution?

Reply:
Also, you wouldn't need to check for nullptr, as the caller already did that

Reply:
> another alternative: passing in Peer& peer. does this mean the lifetime of the reference would just be managed in the normal program execution? so would that mean this is the most efficient?

This is the way to do it. The called function is not taking ownership of `Peer`, so just pass by reference. You only need to pass smart pointers when you're indicating ownership semantics: https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-inout (and items F26/27)

Reply:
👍 done, thanks!